interfaces/smtp/v0/schema.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+"""This file defines the schema for the provider side of the smtp interface.
+
+It exposes one interfaces.schema_base.DataBagSchema subclass called:
+- ProviderSchema
+
+Examples:
+    ProviderSchema:
+        unit: <empty>
+        app: {"smtp":
+                 {
+                    "host": "example.smtp",
+                    "port": "25",
+                    "user": "example_user",
+                    "password_id": "01548499c9233d4612352c989162d940f6a9e6f6d5cc058dfcf66f51575e09c2",
+                    "auth_type": "plain",
+                    "transport_security": "tls",
+                    "domain": "domain",
+                }
+             }
+"""
+from enum import Enum
+from interface_tester.schema_base import DataBagSchema
+from pydantic import BaseModel, Field
+from typing import Optional
+
+
+class TransportSecurity(str, Enum):
+    """Represent the transport security values.
+
+    Attributes:
+        NONE: none
+        STARTTLS: starttls
+        TLS: tls
+    """
+
+    NONE = "none"
+    STARTTLS = "starttls"
+    TLS = "tls"
+
+
+class AuthType(str, Enum):
+    """Represent the auth type values.
+
+    Attributes:
+        NONE: none
+        NOT_PROVIDED: not_provided
+        PLAIN: plain
+    """
+
+    NONE = "none"
+    NOT_PROVIDED = "not_provided"
+    PLAIN = "plain"
+
+class SmtpProviderData(BaseModel):
+    host: str = Field(
+        ...,
+        min_length=1,
+        description="SMTP host.",
+        title="Host",
+        examples=["example.smtp"],
+    )
+    port: int = Field(None,
+        ge=1,
+        le=65536,
+        description="SMTP port.",
+        title="Port",
+        examples=[25],
+    )
+    user: Optional[str] = Field(
+        description="SMTP user.",
+        title="User",
+        examples=["some_user"],
+    )
+    password_id: Optional[str] = Field(
+        description="SMTP password.",
+        title="Password",
+        examples=["somepasswd"],
+    )
+    password_id: Optional[str] = Field(
+        description="Secret ID for the SMTP password.",
+        title="Password ID",
+        examples=["01548499c9233d4612352c989162d940f6a9e6f6d5cc058dfcf66f51575e09c2"],
+    )
+    auth_type: AuthType = Field(
+        description="The type used to authenticate with the SMTP relay.",
+        title="Auth type",
+        examples=[AuthType.NONE],
+    )
+    transport_security: TransportSecurity = Field(
+        description="The security protocol to use for the outgoing SMTP relay.",
+        title="Transport security",
+        examples=[TransportSecurity.NONE],
+    )
+    domain: Optional[str] = Field(
+        description="The domain used by the sent emails from SMTP relay.",
+        title="Domain",
+        examples=["domain"],
+    )
+
+
+class ProviderSchema(DataBagSchema):
+    """Provider schema for SMTP."""
+    app: SmtpProviderData
+
+class RequirerSchema(DataBagSchema):
```

Comment:
I _think_ you can omit this class altogether and the collector won't complain. 
You can locally run `interface_tester discover --include smtp` to verify that the code you wrote is picked up correctly.

You should see something like:

```yaml
  - v0:
   - provider:
      - test_name_1
      - test_name_2
     - schema OK
     - charms:
       - charm-name (https://github.com/canonical/charm-url) custom_test_setup=no
   - requirer:
     - <no tests>
     - schema OK
     - <no charms>
```

Reply:
I'm afraid it's needed
```
Failed to load ProviderSchema from interfaces/smtp/v0/schema.py: schema not defined for role: provider.
Failed to load RequirerSchema from interfaces/smtp/v0/schema.py: schema not defined for role: requirer.

```